pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -666,26 +666,40 @@ public SuccessResponse deleteSegment(
   @Path("/segments/{tableName}")
   @Authorize(targetType = TargetType.TABLE, paramName = "tableName", action = Actions.Table.DELETE_SEGMENT)
   @Authenticate(AccessType.DELETE)
-  @ApiOperation(value = "Delete all segments", notes = "Delete all segments")
-  public SuccessResponse deleteAllSegments(
+  @ApiOperation(value = "Delete the list of segments provided in the payload else all segments",
```

Comment:
Please also update the api doc accordingly

Reply:
Already updated. Do we want to add anything else?

Reply:
We no longer take payload, we take query param instead